src/services/teiid.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * This file is part of ciboard-server
+
+ * Copyright (c) 2022 Andrei Stepanov <astepano@redhat.com>
+ * 
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+/*
+ * https://mongodb.github.io/node-mongodb-native/
+ */
+
+import _ from 'lodash';
+import process from 'process';
+import debug from 'debug';
+import assert from 'assert';
+import { getcfg, TKnownType, known_types } from '../cfg';
+
+const log = debug('osci:teiid');
+const cfg = getcfg();
+
+import { Client, ClientConfig } from 'pg';
+
+export async function _getClient(): Promise<Client | undefined> {
+  /*
+   * pg-client can use environment variables.
+   * Every field of the config object is entirely optional. A Client instance will use environment variables for all missing values.
+   */
+  const config: ClientConfig = cfg.teiid.client_config;
+  let client;
+  try {
+    client = new Client(config);
+    await client.connect();
+  } catch (error) {
+    if (_.isError(error)) {
+      log(' [w] Teiid connection: ', error.message);
+    }
+  }
+  return client;
+}
+
+export const getTeiidClient = _.memoize(_getClient);
+
+(async function init() {
+  /* Query for testing purpose to check on startup if Teiid is responsive */
+  const testQuery = `
+  SELECT
+        c.id, c.created_at, c.fulladvisory
+  FROM
+        Errata_public.errata_main c
+  WHERE
+	      c.id=(SELECT MAX(id) FROM Errata_public.errata_main)
```

Comment:
There's a tab here that should presumably be a space:
```suggestion
        c.id=(SELECT MAX(id) FROM Errata_public.errata_main)
```